troposphere/static/js/components/modals/instance/InstanceLaunchWizardModal.jsx

```diff
@@ -628,6 +628,36 @@ export default React.createClass({
             });
         }
 
+        if (providerSizeList && imageVersion) {
+            let machines = imageVersion.get('machines'),
+                selectedMachine = machines.find(m => m.provider.id == provider.id),
+                limit_size = selectedMachine.size_gb;
+            providerSizeList.forEach(function(size) {
+                let disk_size = size.get('disk');
+                let is_disabled = true;
+                if (disk_size == 0 || limit_size == 0) {
+                    is_disabled = false;
+                } else {
+                    is_disabled = (limit_size > disk_size);
+                //console.log("limit_size "+limit_size+" > disk_size "+disk_size+" = " + is_disabled);
+                }
+                size.set({disabled: is_disabled});
+            });
+            let availableSizes = providerSizeList.cfilter(size=> size.get('disabled') == false),
+                is_available = availableSizes.find(size => (size.id == providerSize.id));
+            if (!is_available) {
+                let newProviderSize;
+                if(!availableSizes) {
+                    newProviderSize = null;
+                    //console.log("Size "+providerSize.get('name')+" no longer available, all other sizes are disabled!");
+                } else {
+                    newProviderSize = availableSizes.first();
+                    //console.log("Size "+providerSize.get('name')+" no longer available, auto-selecting next largest size: "+newProviderSize.get('name'));
+                }
+                providerSize = newProviderSize;
```

Comment:
Changing a state variable is a red flag. This changeset in plain English is saying "Sometimes our state.providerSize is totally wrong so render the right one", but that doesn't change the actual problem that setState is being called with the incorrect value. Calls that use the proper `this.state.providerSize` like `updateState` will get the incorrect value. 

So it's like covering up a state issue, by rendering it to be correct, but allowing it to be wrong elsewhere.